app/wama.web/Controllers/UserToolController.cs

```diff
@@ -236,7 +236,8 @@ public IActionResult AddCertification(string memberId)
         public async Task<IActionResult> ViewCertifications(string memberId)
         {
             SetActiveConsoleTool(Constants.ADMIN_CONSOLE_USERS_PATRONS);
-            var Certifications = await _CertificationService.GetCertificationsAsync(memberId);
+            var Certifications = await _UserAccountService.GetUserAccountAsync(memberId);
+            ViewData["Certifications"] = await _CertificationService.GetCertificationsAsync(memberId);
```

Comment:
Being that we are in the **_ViewCertifications_** action, one would expect that this will return a view that takes a collection of **_Certification_**.

But, you seem to be passing a member to the view instead. This isn't the best software engineering. 

If all you need is a set of extra values that will help in generating your view, then that's what the **_ViewBag_**, or **_ViewData_** is for. The **_model_** parameter you pass to **_View_** should be the main object of interest. In this case, a collection of **_Certification_**.